tests/lib/tools/tests.backup

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: tests.backup prepare"
+    echo "       tests.backup restore"
+}
+
+cmd_prepare() {
+    local BACKUP_PATH=${1:-$(pwd)}
+
+    if [ ! -d "$BACKUP_PATH" ]; then
+        echo "tests.backup: cannot prepare, backup dir does not exist" >&2
+        exit 1
+    fi
+    tar cf "${BACKUP_PATH}.tar" "$BACKUP_PATH"
+}
+
+cmd_restore() {
+    local BACKUP_PATH=${1:-$(pwd)}
+    if [ -f "${BACKUP_PATH}.tar" ]; then
+        find "${BACKUP_PATH}" -maxdepth 1 -mindepth 1 -exec rm -rf {} \;
+        tar -C/ -xf "${BACKUP_PATH}.tar"
+        rm "${BACKUP_PATH}.tar"
+
+        # $BACKUP_PATH was cleaned and recreated, enter the directory again
+        cd "$BACKUP_PATH"
+    else
+        echo "tests.backup: cannot restore, backup file does not exist" >&2
+        exit 1
+    fi
+}
+
+if [ $# -eq 0 ]; then
+    show_help
+    exit
+fi
+
+while [ $# -gt 0 ]; do
+    case "$1" in
+        -h|--help)
+            show_help
+            exit
+            ;;
+        prepare)
+            shift
+            cmd_prepare "$@"
+            exit
+            ;;
+        restore)
+            shift
+            cmd_restore "$@"
+            exit
+            ;;
+        -*)
```

Comment:
Some of the recent tools are not consistent on parsing, not a bloker for this branch but if you have some spare cycles it would be good to unify that.

Reply:
agree, I'll create a pr to unify all the tools as much as possible